api/app/guestRegistrations/api.py

```diff
@@ -0,0 +1,307 @@
+        args = self.req_parser.parse_args()
+        try:
+            user_id = verify_token(request.headers.get('Authorization'))['id']
+            registration = db.session.query(GuestRegistration).filter(
+                GuestRegistration.id == args['guest_registration_id']).first()
+            if registration is None:
+                return 'Registration not found', 404
+            registration.registration_form_id = args['registration_form_id']
+            db.session.commit()
+
+            for answer_args in args['answers']:
+                answer = db.session.query(GuestRegistrationAnswer).filter(
+                    GuestRegistrationAnswer.registration_question_id
+                    == answer_args['registration_question_id']).first()
+                if answer is not None:
+                    answer.value = answer_args['value']
+
+                elif db.session.query(RegistrationQuestion).filter(
+                        RegistrationQuestion.id == answer_args['registration_question_id']).first():
+
+                    answer = GuestRegistrationAnswer(guest_registration_id=registration.id,
+                                                     registration_question_id=answer_args['registration_question_id'],
+                                                     value=answer_args['value'])
+
+                    db.session.add(answer)
+            db.session.commit()
+            return 200
+        except Exception as e:
+            return 'Could not access DB', 400
+
+    def send_confirmation(self, user, questions, answers, confirmed, event_name):
+        if answers is None:
+            LOGGER.warn(
+                'Found no answers associated with response with id {response_id}'.format(response_id=user.id))
+        if questions is None:
+            LOGGER.warn(
+                'Found no questions associated with application form with id {form_id}'.format(form_id=user.id))
+        try:
+            # Building the summary, where the summary is a dictionary whose key is the question headline, and the value
+            # is the relevant answer
+            summary = ""
+            for answer in answers:
+                for question in questions:
+                    if answer.registration_question_id == question.id:
+                        summary += "Question heading :" + question.headline + "\nQuestion Description :" + \
+                                   question.description + "\nAnswer :" + _get_answer_value(
+                            answer, question) + "\n"
+
+            subject = event_name + ' Registration'
+            greeting = strings.build_response_email_greeting(user.user_title, user.firstname, user.lastname)
+            if len(summary) <= 0:
+                summary = '\nNo valid questions were answered'
+            body_text = greeting + self.get_confirmed_message(confirmed) + '\n\n' + summary
+
+            emailer.send_mail(user.email, subject, body_text=body_text)
```

Comment:
Relating to the comment above in the `api/app/guestRegistrations/api.py` file. We need this function to return an indication of successful mail. If this doesn't throw an exception, does it mean that the mail was send? In that case we can just check if this throws an exception and if it doesn't, we assume it was successful.

Reply:
Sorry, this was my fault. I gave them a method signature and my brain was in golang mode. 
golang has a bizarre error system

Reply:
Returned as required